### DISCUSSION TITLE
Feature Request:  Allow hasMany on an upload field

### DISCUSSION THREAD

**Discussion by mo7orh3ad** [payloadcms/payload#130] [Feature Requests & Ideas]
# Bug Report

hasMany does not seem to be assignable to to the field type "upload".

## Expected Behavior

According to the docs (https://payloadcms.com/docs/fields/upload) hasMany should be available for the uploads field.

## Current Behavior

Error: Object literal may only specify known properties, and 'hasMany' does not exist in type 'UploadField'.

## Detailed Description

```
{
    label: 'Gallery',
    name: 'gallery',
    type: 'group',
    fields: [
        {
            label: 'Add images',
            name: 'galleryImage',
            type: 'upload',
            relationTo: 'gallery-images',
            hasMany: true,
        } 
    ]
}, 
```

**Comment by jmikrut** [comment]
Hey @mo7orh3ad — this is actually just an error with the docs. Uploads are not meant to support multiple, but this could potentially be a future enhancement if you would find value out of it.

If you want to support multiple uploads, you could nest an `upload` field within an `array` field and that would also do the trick. That's what we do when we need data like sliders of images, galleries, etc. 

Thoughts?

**Comment by mo7orh3ad** [comment]
Hi @jmikrut, thanks for getting back - using an array field would have been my next approach. I agree, it would be great if this enhancement would find its way into a future release.

**Comment by jmikrut** [comment]
Docs have been updated. We'll keep this one open for now until we publish our roadmap—at which time we will be moving all enhancement issues to the Roadmap instead of tracking them here.

**Comment by denolfe** [comment]
Converted from an issue to a Feature Request discussion thread.

**Comment by slavanossar** [comment]
Would also like to see this, it's a very common use-case to be able to select multiple images and reorder them (galleries, sliders, etc.), using an `array` field with contained `upload` field is very clunky.

Craft's Asset field does this well:
![craft-asset-field](https://user-images.githubusercontent.com/7077600/189258585-f2cf907b-3fe5-4a12-a5f1-ca6d331cd9e8.gif)

**Comment by henrituan** [comment]
Hi, I'd like to know what is the status of this feature request ? I want to contribute to this feature as it's something I need for my project.

Best regard

**Comment by silveltman** [comment]
+1 for this feature.

On top of the improved UX i think it makes sense too in terms of api/json structure, because a uploads field in basically a relationship field with a different UI and some extra settings, right?
